### PR TITLE
instructorEvals: give an option to disable 'not sure' option #1811

### LINKED ISSUE #1811: instructorEvals: give an option to disable 'not sure' option
Requested by a user

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #1811

### CODE REVIEW

**Comment by wkurniawan07** [comment]
Still doing some acceptance testing. I'll signal once ready for review.

**Comment by wkurniawan07** [comment]
Alright, this one is ready for review. No problem occurs when contribution question from old data (no `isNotSureAllowed` field) is used.

File: src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java [hunk 1]
```diff
@@ -18,19 +19,32 @@
...
         super(FeedbackQuestionType.CONTRIB);
+        this.isNotSureAllowed = true;
     }
 
     public FeedbackContributionQuestionDetails(String questionText) {
         super(FeedbackQuestionType.CONTRIB, questionText);
+        this.isNotSureAllowed = true;
+    }
+    
+    private void setContributionQuestionDetails(boolean isNotSureAllowed) {
```
Comment (LowWeiLin) [inline src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java]: Rename this function? It doesn't seem to be setting ContributionQuestionDetails

File: src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java [hunk 2]
```diff
@@ -18,19 +19,32 @@
...
     @Override
     public boolean extractQuestionDetails(
             Map<String, String[]> requestParameters,
             FeedbackQuestionType questionType) {
-        // Nothing to do here.
+        String isNotSureAllowedString = HttpRequestHelper.getValueFromParamMap(
+                requestParameters,
+                Const.ParamsNames.FEEDBACK_QUESTION_CONTRIBISNOTSUREALLOWED);
+        Boolean isNotSureAllowed = (isNotSureAllowedString == null) ? false
+                : (isNotSureAllowedString.equals("on") ? true : false);
```
Comment (LowWeiLin) [inline src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java]: Can use if else statements to make this more readable.

File: src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java [hunk 3]
```diff
@@ -87,12 +102,21 @@ public String getQuestionWithoutExistingResponseSubmissionFormHtml(
 
     @Override
     public String getQuestionSpecificEditFormHtml(int questionNumber) {
-        return "";
+        return FeedbackQuestionFormTemplates.populateTemplate(
+                FeedbackQuestionFormTemplates.CONTRIB_EDIT_FORM,
+                "${questionNumber}", Integer.toString(questionNumber),
+                "${allowingNotSureOrNot}", (isNotSureAllowed) ? "checked=\"checked\"" : "",
```
Comment (LowWeiLin) [inline src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java]: ${allowingNotSureOrNot} -> ${isNotSureAllowedChecked}

**Comment by LowWeiLin** [comment]
Looks good, just some minor changes requested.

**Comment by wkurniawan07** [comment]
The last two are doable. The first one though, I named it `setContributionQuestionDetails` to be consistent with the same operation for other question types.

**Comment by LowWeiLin** [comment]
Ah okay, didn't realize that since there was only one variable to set.

**Comment by wkurniawan07** [comment]
Done and dev green!

**Pull request merged**